docs/rfc/0001-exporter-interface.md

```diff
@@ -0,0 +1,219 @@
+within the Indexer repository, as well as formats and configurations for external plugins and how they will be
+integrated into an Indexer's data pipeline at runtime.
+
+## Problem Statement
+
+[problem-statement]: #problem-statement
+
+Users of Indexer want a way to easily output block data to one or more destinations. This may include standard databases
+such as cloud-managed databases, self-managed SQL clusters, a streaming platform such as Kafka, etc. It may also include
+more complex destinations such as indexing Algorand data via [The Graph](https://github.com/graphprotocol/indexer), or a hosted GraphQL platform such as
+[Fauna](https://github.com/fauna).
+
+The current implementation of Indexer has a single output (which can optionally be disabled) which is heavily tied to
+Postgresql. While there can potentially be many exporter implementations created in the repo which use the existing
+interface and extend functionality to new data outputs, we will be unable to implement all of them and maintain them
+adequately. And any complex data outputs such as those mentioned above will be blocked--users cannot easily use the
+existing interface which Indexer exposes internally to customize their data output.
+
+Once we have a fully implemented exporter plugin interface, users of Indexer will be able to easily develop and
+run plugins which encapsulate their own business logic for data destinations. Some of these will be generic enough that
+we will maintain them centrally in our Indexer repository. Others will be specific to a user's needs and will be
+maintained either publicly or privately  by them. This plugin specification will also give us a runtime framework and
+configuration necessary for creating other plugin systems, such as input plugins and intermediate plugins.
+
+## Design proposal
+
+[design-proposal]: #design-proposal
+
+### Indexer-Native Exporter Plugins  
+Exporter plugins that are native to the Indexer (maintained within the Indexer repository) will necessarily behave
+differently than those which are maintained and built separately and loaded into the pipeline at runtime. The following
+section will outline the properties of so called "Indexer-native" plugins, which will then be extended in the subsequent
+section to incorporate non-native plugins.
+
+***Plugin Interface***  
+Exporter plugins that are native to the Indexer (maintained within the Indexer repository) will each implementation the exporter interface:
+```Go
+package exporters
+
+// ExporterConfig will act as a placeholder for now. It will end up providing an interface for
+// serialization/deserialization of config files.
+// Derived types will provide plugin-specific data fields.
+type ExporterConfig interface {}
+
+
+// Exporter defines the methods invoked during the plugin lifecycle of a data exporter.
+type Exporter interface { 
+	// Name is a UID for each Exporter 
+	Name() string
```

Comment:
There are some mixed tabs and spaces.

Reply:
This should be replaced by a Metadata object containing name, description, deprecation status as we discussed.